sesman/session.c

```diff
@@ -818,14 +864,20 @@ session_start_fork(tbus data, tui8 type, struct SCP_CONNECTION *c,
                 LOG(LOG_LEVEL_INFO, "waiting for window manager "
                     "(pid %d) to exit", window_manager_pid);
                 wm_wait_time = g_time1();
-                g_waitpid(window_manager_pid);
+                wm_exit_code = g_waitpid_status(window_manager_pid);
                 wm_wait_time = g_time1() - wm_wait_time;
-                if (wm_wait_time < 10)
+                if (wm_exit_code != 0)
+                {
+                    LOG(LOG_LEVEL_WARNING, "window manager (pid %d) exited with "
+                        "non-zero exit code (%d). Window manager config problem?",
+                        window_manager_pid, wm_exit_code);
+                }
+                else if (wm_wait_time < 10)
```

Comment:
Suggest making the `wm_wait_time < 10` check unconditional, as this was explicitly introduced to catch the common case of the session exiting early. This is something I generally end up triaging, so I was hoping to be able to check logs for 'exited quickly'. Your change makes this harder to spot. How about something like:-

```c
wm_exit_code = g_waitpid_status(window_manager_pid);
wm_wait_time = g_time1() - wm_wait_time;
if (wm_exit_code != 0)
{
    LOG(LOG_LEVEL_WARNING, "window manager (pid %d) exited with "
             "non-zero exit code (%d)",
             window_manager_pid, wm_exit_code);
}
if (wm_wait_time < 10)
{
    /* Common config issue */
    LOG(LOG_LEVEL_WARNING, "window manager (pid %d) exited quickly "
             "(%d secs). Window manager config problem?",
             window_manager_pid, wm_wait_time);
}

```

Reply:
done